### DISCUSSION TITLE
Get complete Eigenvectors and Eigenvalues

### DISCUSSION THREAD

**Discussion by hcut** [QuSpin/QuSpin#615] [Q&A]
I am trying to find the eigenvalue/vectors of 2 particle spin Heisenberg hamiltonian H. Eigenvectors/values should be in number 4 as  dim(H)=2^2, but I am getting only 2. Please help me out here.

Here it is: 

     from __future__ import print_function, division
     import sys,os
     from quspin.operators import hamiltonian,quantum_operator 
     from quspin.basis import spin_basis_1d 
     import numpy as np 
     L=2
     basis = spin_basis_1d(L,S='1/2',pauli=False,Nup=L//2)
     #j_pm_list = [[1.0/2.0,i,(i+1)%L] for i in range(L)]
      j_pm_list = [[1.0/2.0,i,(i+1)] for i in range(L-1)]
      j_mp_list=j_pm_list
      #j_zz_list = [[1.0,i,(i+1)%L] for i in range(L)]
      j_zz_list = [[1.0,i,(i+1)] for i in range(L-1)]
      op_list_0=[["+-",j_pm_list],["-+",j_mp_list],["zz",j_zz_list]] 
      op_dict=dict(H0=op_list_0)
      H=quantum_operator(op_dict,basis=basis,dtype=np.float32,check_symm=False)
      E,V = H.eigh(pars=dict(H0=1.0))   
      print(E)
      print(V)

output: 
[-0.75  0.25]
[[ 0.70710677  0.70710677]
 [-0.70710677  0.70710677]]

**Comment by andyjaj** [comment]
In your basis = ... line you have fixed Nup=L//2. That is, you've requested that the total z magnetisation, m_z= 0. This has restricted the Hilbert space to just two states.
